test/junit/scala/collection/immutable/StreamTest.scala

```diff
@@ -15,4 +18,58 @@ class StreamTest {
     assertTrue(Stream(1,2,3,4,5).filter(_ < 4) == Seq(1,2,3))
     assertTrue(Stream(1,2,3,4,5).filterNot(_ > 4) == Seq(1,2,3,4))
   }
+
+  /** Test helper to verify that the given Stream operation allows
+    * GC of the head during processing of the tail.
+    */
+  def assertStreamOpAllowsGC[B](op: (=> Stream[Int], Int => B) => Any, f: Int => B): Unit = {
+    val msgSuccessGC = "GC success"
+    val msgFailureGC = "GC failure"
+
+    val ref = WeakReference( Stream.from(1).take(500) ) // wait about 5 seconds to fail
+
+    def gcAndThrowIfCollected(n: Int): B = {
+      System.gc()
+      Thread.sleep(10)
+      if (ref.get.isEmpty) throw new RuntimeException(msgSuccessGC)
+      assertTrue(n >= 1) // protect from JIT optimizations
```

Comment:
Perhaps we can remove that line now. It was added in response to an earlier suggestion, I've forgotten from whom. However, the inlining that ultimately was the issue was at the call to `.withFilter`, and the `@noinline` was the fix, so this may be doing nothing?